tbdex/tbdex.go

```diff
@@ -0,0 +1,125 @@
+package tbdex
+
+import (
+	"encoding/json"
+	"fmt"
+
+	libclose "github.com/TBD54566975/tbdex-go/tbdex/closemsg"
+	"github.com/TBD54566975/tbdex-go/tbdex/message"
+	liborder "github.com/TBD54566975/tbdex-go/tbdex/order"
+	libquote "github.com/TBD54566975/tbdex-go/tbdex/quote"
+	librfq "github.com/TBD54566975/tbdex-go/tbdex/rfq"
+)
+
+// Message is the interface that all tbdex messages implement. Especially useful for decoding and parsing messages
+// when the kind of message is not known upfront.
+type Message interface {
+	ValidNext() []string
+	Kind() string
+	Digest() ([]byte, error)
+	// TODO: uncomment these once rfq has been refactored to separate privateStrict bool
+	// Verify() error
+	// Parse([]byte) (Message, error)
+}
+
+type msg struct {
+	Metadata message.Metadata
+}
+
+// DecodeMessage unmarshals a message. It uses the metadata kind to determine the type of message.
+//
+// # Note
+//
+// unmarshaling includes validation
+func DecodeMessage(input []byte) (Message, error) {
```

Comment:
Nit: Why call `decode` instead of `unmarshal`?

Reply:
Good point. Can do will do! Was sad that I couldn't use `UnmarshalJSON` on an interface but made sense as to why

Reply:
done! https://github.com/TBD54566975/tbdex-go/commit/f791e147fd90daa88eb5896f8646e27571bf817b